src/marshmallow/fields.py

```diff
@@ -1855,6 +1858,73 @@ class IPv6Interface(IPInterface):
     DESERIALIZATION_CLASS = ipaddress.IPv6Interface
 
 
+class Enum(String):
+
+    default_error_messages = {
+        "unknown": "Must be one of: {choices}.",
+    }
+
+    def __init__(
+        self,
+        enum,
+        *args,
+        **kwargs,
+    ):
+        self.enum = enum
+        self.choices = ", ".join(enum.__members__)
+        super().__init__(*args, **kwargs)
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if value is None:
+            return None
+        return value.name
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        value = super()._deserialize(value, attr, data, **kwargs)
+        try:
+            return getattr(self.enum, value)
+        except AttributeError as exc:
+            raise self.make_error("unknown", choices=self.choices) from exc
+
+
+class TypedEnum:
+    """Base class for typed Enum fields"""
+
+    default_error_messages = {
+        "unknown": "Must be one of: {choices}.",
+    }
+
+    def __init__(
+        self,
+        enum,
+        *args,
+        **kwargs,
+    ):
+        self.enum = enum
+        self.choices = ", ".join([str(m.value) for m in enum])
+        super().__init__(*args, **kwargs)
+
+    def _serialize(self, value, attr, obj, **kwargs):
+        if value is None:
+            return None
+        return value.value
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        value = super()._deserialize(value, attr, data, **kwargs)
+        try:
+            return self.enum(value)
+        except ValueError as exc:
+            raise self.make_error("unknown", choices=self.choices) from exc
+
+
+class StringEnum(TypedEnum, String):
```

Comment:
Rather than `str(value)`, which would expose the enum name, I'd rather use the member name (`SymbolEnum`). Offering to serialize by value allows users to provide human-readable names as enum values, for instance, or at least decouple business logic and API layer a bit.

But thanks for the feedback. Your comment confirms my intuition that the dump/load asymmetry is too much API surface for the need.

Reply:
Yeah the symbol name is what I meant, just worded it kind of poorly. The asymmetry was a pain in the ass for tests and it tripped me up a few times. Pretending that they're just a union of string literals is the way to go imo